django/urls/resolvers.py

```diff
@@ -90,6 +90,11 @@ def __init__(self, regex):
         # expression.
         self._regex = regex
         self._regex_dict = {}
+        # As a performance optimization, if the given regex string is a regular
+        # string (not a lazily-translated string proxy), we can just compile it
+        # once and avoid the property and the per-langage compilation entirely.
+        if isinstance(regex, six.string_types):
+            self.__dict__['regex'] = self._compile(regex)
```

Comment:
Setting `self.__dict__['regex']` doesn't work with `@property` (tested on 2.7). I don't think it's even possible to override the return value of a property like this. You'll have to dynamically return the right value in the `regex` getter.